kernel/src/engine/arrow_expression.rs

```diff
@@ -147,41 +147,28 @@ impl ProvidesColumnByName for StructArray {
     }
 }
 
-fn extract_column<'array, 'path>(
-    array: &'array dyn ProvidesColumnByName,
-    path_step: &str,
-    remaining_path_steps: &mut impl Iterator<Item = &'path str>,
-) -> Result<&'array Arc<dyn Array>, ArrowError> {
-    let child = array
-        .column_by_name(path_step)
-        .ok_or(ArrowError::SchemaError(format!(
-            "No such field: {}",
-            path_step,
-        )))?;
-    if let Some(next_path_step) = remaining_path_steps.next() {
-        // This is not the last path step. Drill deeper.
-        extract_column(
-            column_as_struct(path_step, &Some(child))?,
-            next_path_step,
-            remaining_path_steps,
-        )
-    } else {
-        // Last path step. Return it.
-        Ok(child)
+fn extract_column<'a>(
```

Comment:
maybe a super quick example would also be useful? e.g. parent is complex struct column `a` and field names are `[b, c]` means we return a ref to the nested column `c`? am I understanding correctly?

Reply:
Exactly. Added the missing doc comment.